scripts/zones/Western_Adoulin/npcs/Sylvie.lua

```diff
@@ -0,0 +1,102 @@
+-----------------------------------
+--  Area: Western Adoulin (256)
+--   NPC: Sylvie
+--  Type: NPC
+--  Starts Dances with Luopans
+--  !pos 78.094 32.000 135.725
+-----------------------------------
+require("scripts/globals/keyitems")
+require("scripts/globals/npc_util")
+require("scripts/globals/quests")
+require("scripts/globals/status")
+local ID = require("scripts/zones/Western_Adoulin/IDs")
+
+function onTrade(player,npc,trade)
+    -- DANCES WITH LUOPANS
+    if player:getQuestStatus(ADOULIN, tpz.quest.id.adoulin.DANCES_WITH_LUOPANS) == QUEST_ACCEPTED then
+        if player:hasKeyItem(tpz.ki.FISTFUL_OF_HOMELAND_SOIL) and npcUtil.tradeHas(trade, 703) then -- Petrified Log
+            player:startEvent(34)
+        end
+    end
+end
+
+function onTrigger(player,npc)
+    -- Buying a replacement Matre Bell on Geomancer
+    if player:getLocalVar("Sylvie_Need_Zone") == 0 and player:getMainJob() == tpz.job.GEO and not player:hasItem(21460) then  -- Matre Bell
+        player:setLocalVar("Sylvie_Need_Zone", 1)
+        player:startEvent(37)
+        return
+    end
+
+    -- DANCES WITH LUOPANS
+    local dwlQuestStatus = player:getQuestStatus(ADOULIN, tpz.quest.id.adoulin.DANCES_WITH_LUOPANS)
+    if dwlQuestStatus == QUEST_COMPLETED then
+        player:startEvent(39)
+    elseif player:getCharVar("GEO_DWL_Luopan") == 1 then
+        player:startEvent(36)
+    elseif dwlQuestStatus == QUEST_ACCEPTED and player:hasKeyItem(tpz.ki.LUOPAN) then
+        player:startEvent(35)
+    elseif dwlQuestStatus == QUEST_ACCEPTED then
+        player:startEvent(33)
+    elseif player:getCharVar("GEO_DWL_Triggered") == 1 then
+        player:startEvent(32)
+    elseif dwlQuestStatus == QUEST_AVAILABLE and player:getMainLvl() >= ADVANCED_JOB_LEVEL then
+        player:startEvent(31)
+    else
+        player:startEvent(38)  -- Standard dialog
+    end
+end
+
+function onEventUpdate(player,csid,option)
+    -- Buying a replacement Matre Bell on Geomancer
+    if csid == 37 then
+        if option == 1 then  -- gil
+            if player:getGil() >= 300000 then
+                player:setLocalVar("Sylvie_Matre_Bell", 1)
+                player:updateEvent(7, 1, 2964, 0, 0, 0, 4480, 1)
+            end
+        elseif option == 2 then  -- bayld
+            if player:getCurrency('bayld') >= 150000 then
+                player:setLocalVar("Sylvie_Matre_Bell", 2)
+                player:updateEvent(256, 0, 127, 0, 0, 0, 4095, 1)
```

Comment:
For updateEvent params, retail will often send "junk data" as params which aren't required for the event (usually random memory values). We usually try to keep params limited to what we know is the cause of different behavior (so it's easier to tell which params are "important" when debugging in the future).

I toyed with the EventUpdate locally, and param 8 is what determines Sylvie's next line for a chosen option:
```
0 - You can't afford it
1 - Okay here you go
2 - You don't have the inventory space
```

I was a little surprised to find the special inventory space line. While doing a `npcUtil.giveItem` check is what one _usually_ wants to do, Sylvie is special.

Thinking something like:
```lua
local updateParam = 0
if player:getFreeSlotsCount() >= 1 then
    if option == 1 and player:getGil() >= 300000 then
        player:setLocalVar("Sylvie_Matre_Bell", 1)
        updateParam = 1
    elseif option == 2 and player:getCurrency('bayld') >= 150000 then
        player:setLocalVar("Sylvie_Matre_Bell", 2)
        updateParam = 1
    end
else
    updateParam = 2
end

player:updateEvent(0, 0, 0, 0, 0, 0, 0, updateParam)
```

You can still keep `giveItem` inside the `if` for the `onFinish`; you'll need it to give the item anyway!

Reply:
Ah,  I see. I just mirrored the exact params for updateEvent from the captures, to be safe.

I'm happy to clean that up, as I think it simplifies the code too by removing the rest of the seemingly magic numbers.

After work I can try to apply this change and play around with it to verify that it works for all edge cases.

Reply:
Rewrote this. Those params worked great. Please take a look.